types/unitparser_test.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAsSIHappy(t *testing.T) {
+	cases := []struct {
+		in, unit string
+		exp      int64
+	}{
+		{"1", "", 1},
+		{"1 km", "m", 1000},
+		{"1KB", "B", 1000},
+		{"2 KiB", "B", 2048},
+		{"1MJ", "J", 1000000},
+		{"2 MiB", "B", 2 * 1024 * 1024},
+	}
+
+	for _, c := range cases {
```

Comment:
Let's use sub-tests parameterized with `in` and `unit` so that we can easily see which test case fails.